src/main/plugins/org.dita.pdf2/xsl/common/topicmergeImpl.xsl

```diff
@@ -46,26 +46,108 @@ See the accompanying LICENSE file for applicable license.
+    
+    <xsl:variable name="originalMap" as="element()">
+        <xsl:sequence select="/dita-merge/*[contains(@class,' map/map ')][1]"/>
+    </xsl:variable>
 
     <xsl:output indent="no"/>
 
-    <xsl:key name="topic" match="dita-merge/*[contains(@class,' topic/topic ')]" use="concat('#',@id)"/>
-    <xsl:key name="topic" match="dita-merge/dita" use="concat('#',*[contains(@class, ' topic/topic ')][1]/@id)"/>
+    <xsl:key name="topic" match="dita-merge/*[contains(@class,' topic/topic ')]|dita-merge/dita/*" use="concat('#',@id)"/>
     <xsl:key name="topicref" match="//*[contains(@class,' map/topicref ')]" use="generate-id()"/>
 
-<!--
   <xsl:template match="/">
-    <xsl:copy-of select="."/>
+      <xsl:variable name="updatedMap"  as="document-node()">
+          <xsl:document>
+          <xsl:choose>
+              <xsl:when test="/dita-merge/dita">
+                  <xsl:apply-templates select="/dita-merge" mode="resolve-root-dita">
+                      <xsl:with-param name="ditaDocs" as="element()">
+                          <root>
+                              <xsl:sequence select="/dita-merge/dita"/>
+                          </root>
+                      </xsl:with-param>
+                  </xsl:apply-templates>
+              </xsl:when>
+              <xsl:otherwise>
+                  <xsl:sequence select="/*"/>
+              </xsl:otherwise>
+          </xsl:choose>
+          </xsl:document>
+      </xsl:variable>
+      <xsl:apply-templates select="$updatedMap/*"/>
   </xsl:template>
--->
 
-  <xsl:template match="dita-merge">
+
+    <xsl:template match="dita-merge">
         <xsl:variable name="map" select="(*[contains(@class,' map/map ')])[1]"/>
         <xsl:element name="{name($map)}">
           <xsl:copy-of select="$map/@*"/>
           <xsl:apply-templates select="$map" mode="build-tree"/>
         </xsl:element>
     </xsl:template>
+    
+    <xsl:template match="*[contains(@class,' map/topicref ')][@first_topic_id]" mode="resolve-root-dita">
+        <xsl:param name="ditaDocs" as="element()+"/>
+        <xsl:variable name="topicId" select="substring-after(@first_topic_id,'#')"/>
+        <xsl:choose>
+            <xsl:when test="$ditaDocs/*[@id=$topicId]">
+                <xsl:variable name="topicref" select="." as="element()"/>
+                <xsl:for-each select="$ditaDocs/*[@id=$topicId]/*[contains(@class,' topic/topic ')]">
+                    <xsl:variable name="updateTopicTarget" select="concat('#',@id)"/>
+                    <xsl:variable name="lastTopic" select="if (position()=last()) then true() else false()" as="xs:boolean"/>
+                    <xsl:for-each select="$topicref">
+                        <xsl:copy>
+                            <xsl:copy-of select="@*[not(name()='href' or name()='first_topic_id')]"/>
+                            <xsl:attribute name="first_topic_id"><xsl:value-of select="$updateTopicTarget"/></xsl:attribute>
```

Comment:
Use `<xsl:attribute name="first_topic_id" select="$updateTopicTarget"/>` instead since it's more simple.